snuba/clickhouse/http.py

```diff
@@ -31,20 +31,32 @@ def __repr__(self) -> str:
 
 
 class HTTPBatchWriter(BatchWriter):
-    def __init__(self, schema, host, port, options=None, table_name=None):
+    def __init__(self,
+        schema,
+        host,
+        port,
+        encoder: Callable[[WriterTableRow], Iterable[bytes]],
+        options=None,
+        table_name=None,
+        # if chunk_size is none we let urlopen chunk on its own.
+        chunk_size: Optional[int]=None,
+    ):
         self.__schema = schema
         self.__pool = HTTPConnectionPool(host, port)
         self.__options = options if options is not None else {}
         self.__table_name = table_name or schema.get_table_name()
+        self.__chunk_size = chunk_size
+        self.__encoder = encoder
 
-    def __default(self, value):
-        if isinstance(value, datetime):
-            return value.strftime(DATETIME_FORMAT)
-        else:
-            raise TypeError
-
-    def __encode(self, row: WriterTableRow) -> bytes:
-        return json.dumps(row, default=self.__default).encode("utf-8")
+    def _prepare_body(self, rows: Iterable[WriterTableRow]) -> Iterable[bytes]:
+        ret = []
+        chunk_size = self.__chunk_size or 1
+        for i in range(0, len(rows), chunk_size):
```

Comment:
> snuba/clickhouse/http.py:54: error: Argument 1 to "len" has incompatible type "Iterable[Mapping[str, Any]]"; expected "Sized"
> snuba/clickhouse/http.py:56: error: Value of type "Iterable[Mapping[str, Any]]" is not indexable

This needs to use something like the [`chunked` iterable wrapper in Sentry.](https://github.com/getsentry/sentry/blob/master/src/sentry/utils/iterators.py#L27-L36)